src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -42,19 +54,21 @@ public CompanyProfileService(Logger logger, CompanyProfileRepository companyProf
      */
     public void updateInsolvencyLink(final CompanyProfile companyProfileRequest)
             throws NoSuchElementException {
-        Optional<CompanyProfileDocument> companyProfileOptional = companyProfileRepository
-                .findCompanyProfileDaoByCompanyProfile_Data_CompanyNumber(
-                        companyProfileRequest.getData().getCompanyNumber());
+        String companyNumber = companyProfileRequest.getData().getCompanyNumber();
+        Query updateCriteria = new Query(Criteria.where("data.company_number").is(companyNumber));
+        Update updateQuery = new Update();
+        updateQuery.set("data.links.insolvency",
+                companyProfileRequest.getData().getLinks().getInsolvency());
+        UpdateResult updateResult = mongoTemplate.updateFirst(updateCriteria,
+                updateQuery,
+                "company_profile");
 
-        if (companyProfileOptional.isEmpty()) {
-            throw new NoSuchElementException("Database entry not found");
+        if (updateResult.getModifiedCount() == 1) {
+            logger.trace(String.format("DSND-376: Insolvency links updated for company number: %s",
+                    companyNumber));
+        }
+        if (updateResult.getMatchedCount() == 0) {
```

Comment:
I dont think this approach will work once we start updating multiple attributes only if the links need to be updated

Reply:
I thought we are updating a single document as we are fetching one by its companyNumber (Where companyNumber is its ID too)